pkg/reconciler/ingress/ingress.go

```diff
@@ -213,6 +216,16 @@ func (r *Reconciler) reconcileIngress(ctx context.Context, ing *v1alpha1.Ingress
 		// about the steady state.
 		logger.Debug("Kingress is ready, skipping probe.")
 		ready = true
+	} else if hasStatus, readyStatus := r.virtualServicesReady(ctx, vses); hasStatus {
+		// Check if our VirtualServices have a status property.
```

Comment:
Should this check be first? Probing initialization is not cheap.

Reply:
Which probing initialization are you talking about? This is called before statusManager.IsReady

Reply:
Ah yes, my bad.
`ing.IsReady()` looks a lot like `statusManager.IsReady(ctx, ing)`